_studio/shared/src/libmfx_allocator_vaapi.cpp

```diff
@@ -148,6 +148,11 @@ static void FillSurfaceAttrs(std::vector<VASurfaceAttrib> &attrib, unsigned int
             break;
         case MFX_FOURCC_RGBP:
             format = VA_RT_FORMAT_RGBP;
+            attrib.resize(attrib.size()+1);
+            attrib[1].type            = VASurfaceAttribUsageHint;
+            attrib[1].flags           = VA_SURFACE_ATTRIB_SETTABLE;
+            attrib[1].value.type      = VAGenericValueTypeInteger;
+            attrib[1].value.value.i   = VA_SURFACE_ATTRIB_USAGE_HINT_DECODER;
```

Comment:
I'm not sure that this is correct. Why HINT_DECODER?

Reply:
This Hint is used in media driver for deciding the tiling of surface. @chuanli1 Please give more detail description/comment for this.

Reply:
It's okay. But you just hardcode this info. I mean it's just an assumption this this function is allocating surface for decoder. And the assumption can be wrong.

Reply:
For decode pipeline, I need to make the value of [mediaSurface->surfaceUsageHint equal to VA_SURFACE_ATTRIB_USAGE_HINT_DECODER](https://github.com/intel/media-driver/blob/master/media_driver/linux/common/ddi/media_libva.cpp#L2509), 

[code](https://github.com/intel/media-driver/blob/master/media_driver/linux/common/ddi/media_libva_util.cpp#L231)
For RGBP, the VA_SURFACE_ATTRIB_USAGE_HINT_DECODER flag enable tileformat equal to I915_TILING_Y.

Reply:
memType can‘t act on this [if condition judgment](https://github.com/intel/media-driver/blob/master/media_driver/linux/common/ddi/media_libva_util.cpp#L231)

Reply:
Please see again: you're **unconditionally** adding hardcoded VA_SURFACE_ATTRIB_USAGE_HINT_DECODER **to a common function which has nothing about decode**. Why USAGE_HINT_DECODER? Not USAGE_HINT_ENCODER? Not TRANSCODER?

Reply:
Could we use one another function for decoding only?

Reply:
Again, why can't we do something similar what we have at lines https://github.com/Intel-Media-SDK/MediaSDK/blob/7f372a73ea73ef8a9c2fb853559a2a23a95f3e48/_studio/shared/src/libmfx_allocator_vaapi.cpp#L156-L157

Reply:
Yes. Thanks for your comments. @chuanli1 could you please change it？

Reply:
okay, thanks @wangyan-intel  and @dmitryermilov , I'll change it.